packages/forma-36-react-components/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -129,17 +146,31 @@ export const Autocomplete = <T extends {}>({
   );
 
   const toggleList = (isOpen?: boolean): void => {
-    dispatch({ type: TOGGLED_LIST, payload: isOpen });
+    dispatch({
+      type: TOGGLED_LIST,
+      payload: isOpen,
+      multiselect: !!multiselect,
+    });
   };
 
-  const selectItem = (item: T) => {
-    dispatch({ type: ITEM_SELECTED });
-    onQueryChange('');
+  const selectItem = (item: T, index: number = 0) => {
+    dispatch({
+      type: ITEM_SELECTED,
+      payload: index,
+      multiselect: !!multiselect,
+    });
+    if (!multiselect) {
+      onQueryChange('');
+    }
     onChange(item);
   };
 
   const updateQuery = (value: string) => {
-    dispatch({ type: QUERY_CHANGED, payload: value });
+    dispatch({
+      type: QUERY_CHANGED,
+      payload: value,
+      multiselect: !!multiselect,
```

Comment:
Since `multiselect` is a boolean, I think the double `!!` is unnecessary, and then this would be cleaner to do:

```js
{
  type: QUERY_CHANGED,
  payload: value,
  multiselect,
}
```

What do you think?